web/src/main/kotlin/com/example/ajaxproject/nats/NatsController.kt

```diff
@@ -3,14 +3,22 @@ package com.example.ajaxproject.nats
 import com.google.protobuf.GeneratedMessageV3
 import com.google.protobuf.Parser
 import io.nats.client.Connection
+import io.nats.client.Message
+import reactor.core.publisher.Mono
 
-interface NatsController<RequestT : GeneratedMessageV3, ResponseT : GeneratedMessageV3> {
+interface NatsController<ReqT : GeneratedMessageV3, RespT : GeneratedMessageV3> {
 
     val subject: String
 
     val connection: Connection
 
-    val parser: Parser<RequestT>
+    val parser: Parser<ReqT>
 
-    fun handle(request: RequestT): ResponseT
+    fun generateReplyForNatsRequest(request: ReqT): Mono<RespT>
```

Comment:
This is really bad name.

Your nats controllers don't generate response for requests. The controllers do some useful work.
Generating of a reply is one of the function.

Reply:
In case with my nats changes - i wont to refactor them all with test - after approval of my main service/repo to react.  I plan to go back to the previous structure and this method will disappear.

Reply:
ok